src/pharmpy/tools/ruvsearch/tool.py

```diff
@@ -334,9 +342,16 @@ def _create_combined_model(input_model, current_iteration):
     return model
 
 
-def _create_dataset(input_model):
+def _create_dataset(input_model, dv):
     # Non-observations have already been filtered
     residuals = input_model.modelfit_results.residuals
+    if dv:
+        input_dataset = input_model.dataset
+        input_dataset = input_dataset[input_dataset['DV'] != 0].reset_index(
+            drop=True
+        )  # filter non-observations
+        indices = input_dataset.index[input_dataset['DVID'] == dv].tolist()
```

Comment:
To get the `DVID` column you should use the datainfo of the input model and use the `typeix`

Reply:
The DVID column has type unknown in datainfo for the cases that I have checked.